### PR TITLE
DLT-14910 Fix scalardb_fdw issues incorrectly returning 0 results for particular queries

### PR DESCRIPTION
**Pull request by choplin**
## Description

ScalarDB Analytics with PostgreSQL returns empty results wrongly for particular queries, including the query of [the sample query in the official document](https://scalardb.scalar-labs.com/docs/latest/scalardb-samples/scalardb-analytics-postgresql-sample/README/#run-analytical-queries).

This PR fixes this issue.

## Related issues and/or PRs

https://scalar-labs.atlassian.net/browse/DLT-14910

## Changes made

The `attrs_used`, which is a list of attributes that we ask the foreign query to return, is now retrieved after determining `local_conds`, which is a list of conditions evaluated locally on the PostgreSQL side.

The issue is caused by the column-pruning feature. The expected behavior is that the ForeignScan returns the columns that are included in the target list or required to evaluate expressions in join, where, order by, etc. However, since the `attrs_used` was calculated before the `local_conds` was initialized, the query results became an empty wrongly.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

This issue affects #48 because it also bumps up the `scalardb_fdw` version to 3.10, which introduced the column-pruning feature. 

Thank you @kota2and3kan and @josh-wong for reporting the issue!

## Release notes

Fixes wrong behavior in column pruning in `scalardb_fdw`

### CODE REVIEW

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

File: scalardb_fdw/Makefile
Comment (choplin) [reply scalardb_fdw/Makefile:36]: Ah, sorry. I forgot to leave a comment on this. Right, this is not directly related to this PR. This is required when the JDK distribution has multiple paths to `libjvm`. For example, Azule JDK 8 on macOS provides two paths. (They are physically the same file because `jre` is a symbolic link to `zulu-8.jdk/Contents/Home/jre`.)

```console
jre/lib/server/libjvm.dylib
zulu-8.jdk/Contents/Home/jre/lib/server/libjvm.dylib
```

It is necessary for this Makefile to work properly to set a single path to `libjvm_path.`. That's why we need `word 1` to select a single path.

Previously, I used `adoptopenjdk-8`, which provides only a single path to `libjvm`, so I didn't notice this issue.

File: scalardb_fdw/expected/scalardb_fdw.pg15.out
Comment (choplin) [reply scalardb_fdw/expected/scalardb_fdw.pg15.out:903]: Hmm, right. It might change between the versions. One possible naive approach is to check the Postgres version when the tests start and abort the tests if the version differs from what we expect. 

Also, it would be helpful to fix the version of Postgres to use for the testing using a tool like [mise](https://mise.jdx.dev/). (I use it for the local testing, but don't include the `.mise.toml` in the repository so far.)

But I haven't encountered such a situation so far, so let's leave it as it is for now. Once we see the test failing due to the difference in the cost, we will consider how to solve it then.

**Comment by feeblefakie** [comment]
@choplin Please add me and Toshi as reviewers for ScalarDB-related repos so we can catch up with what's happening. 🙇 
(I don't need to review the code of all PRs, but I would like to catch up from product management perspectives.)

**Review by josh-wong — APPROVED** [review]
Thank you for explaining these changes in real time today! LGTM👍

**Review by kota2and3kan — APPROVED** [review]
LGTM! Thank you for your explanation in the meeting!

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Comment by choplin** [comment]
@feeblefakie Just to confirm, should I wait for your approval for this PR? Or, since we add you as a reviewer to have you catch up, can I merge it without waiting for you?

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by feeblefakie** [comment]
@choplin
> Just to confirm, should I wait for your approval for this PR? 

For improvement/enhancement PRs, please wait for my approval. (if my review is too slow, please ping me.)
For bugfix PRs, it would be helpful if you could wait, but you can merge them if my review is too slow and other reviewers approve. In such a case, it would be helpful if you could leave a comment to ask me to review later like "I'm merging this but PTAL when you get a chance!".

**Comment by choplin** [comment]
@feeblefakie Thank you for your detailed explanation! I'll follow that way next time.

**Pull request merged**
